### PR TITLE
feat(linter): implement react/require-render-return

### PR DESCRIPTION
**Pull request by kaykdm**
implement react/require-render-return
Related issue: https://github.com/oxc-project/oxc/issues/1022

original
- doc: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/require-render-return.md
- code: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/lib/rules/require-render-return.js

### CODE REVIEW

**Comment by XantreGodlike** [comment]
I think you should add help info

File: crates/oxc_linter/src/rules/react/require_render_return.rs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+                                    None
+                                }
+                            }
+                            _ => None,
+                        })
+                        .find(|fn_body| !has_return_in_fn_body(fn_body))
+                    {
+                        ctx.diagnostic(RequireRenderReturnDiagnostic(fn_body.span));
+                    }
+                }
```
Comment (camc314) [inline crates/oxc_linter/src/rules/react/require_render_return.rs:114]: am reviewing fairly quickly, so correct me if i am wrong. But, won't this match

```ts
foo({ render: render: function() {} })
```
?

This is what the plugin reports for me (js not oxc)

Screenshot 2024-01-09 at 22 43 56


can we add a passing test case for the example above (unless i'm wrong lol)

Comment (camc314) [reply crates/oxc_linter/src/rules/react/require_render_return.rs:114]: lol i was wrong - should've read the code more carefully.

above:
```rs
if !is_es5_component(node) && !is_es6_component(node) {
    return;
}
```

`react.rs`
```rs
const PRAGMA: &str = "React";
const CREATE_CLASS: &str = "createReactClass";

pub fn is_es5_component(node: &AstNode) -> bool {
    let AstKind::CallExpression(call_expr) = node.kind() else { return false };

    if let Expression::MemberExpression(member_expr) = &call_expr.callee {
        if let Expression::Identifier(ident) = member_expr.object() {
            return ident.name == PRAGMA
                && member_expr.static_property_name() == Some(CREATE_CLASS);
        }
    }

    if let Some(ident_reference) = call_expr.callee.get_identifier_reference() {
        return ident_reference.name == CREATE_CLASS;
    }

    false
}
```

**Review by camc314 — APPROVED** [review]
Thank you!

**Pull request merged**